Sources/PlaybackSDK/PlayerUIView/PlaybackUIView.swift

```diff
@@ -74,21 +93,30 @@ internal struct PlaybackUIView: View {
     /**
      Loads the HLS stream for the provided entry ID and authorization token.
      
-     This method asynchronously fetches the HLS stream URL using the `PlayBackSDKManager` and updates the `videoURL` state variable accordingly.
+     This method asynchronously fetches the HLS stream URL using the `PlaybackSDKManager` and updates the `videoURL` state variable accordingly.
      */
     private func loadHLSStream() {
-        PlayBackSDKManager.shared.loadHLSStream(forEntryId: entryId, andAuthorizationToken: authorizationToken) { result in
+        
+        //TO-DO Fetch all HLS urls from the entryID array
+        PlaybackSDKManager.shared.loadAllHLSStream(forEntryIds: entryId, andAuthorizationToken: authorizationToken) { result in
             switch result {
-            case .success(let hlsURL):
-                print("HLS URL: \(hlsURL)")
+            case .success(let videoDetails):
                 DispatchQueue.main.async {
-                    self.videoURL = hlsURL
+                    self.videoDetails = videoDetails.0
+                    self.playlistErrors = videoDetails.1
                     self.hasFetchedVideoDetails = true
+                    if (!(self.playlistErrors?.isEmpty ?? false)) {
+                        onError?(self.playlistErrors?.last ?? .unknown)
+                        onErrors?(self.playlistErrors ?? [])
+                    }
                 }
             case .failure(let error):
                 // Trow error to the app
                 onError?(error)
-                print("Error loading HLS stream: \(error)")
+                onErrors?([error])
```

Comment:
We are emitting this error twice because the listener could be both `onError` as well as `onErrors`: `onError` in case the caller is a single video request, `onErrors` in case the caller is a multiple video request for the playlist.